test/src/locator.spec.ts

```diff
@@ -581,6 +581,109 @@ describe('Locator', function () {
     });
   });
 
+  describe('Locator.last', () => {
+    it('should time out when any one of the locaters do not match', async () => {
+      const clock = sinon.useFakeTimers({
+        shouldClearNativeTimers: true,
+        shouldAdvanceTime: true,
+      });
+      try {
+        const {page} = await getTestState();
+
+        await page.setViewport({width: 500, height: 500});
+        await page.setContent(`
+        <button id="test1">test1</button>
+        <input id="test2">test2</input>
+      `);
+        const result = Locator.last([
+          page.locator('#test1'),
+          page.locator('#test2'),
+          page.locator('#test3'),
+        ])
+          .setTimeout(5000)
+          .waitHandle();
+        clock.tick(5100);
+        await expect(result).rejects.toEqual(
+          new TimeoutError('Timed out after waiting 5000ms')
+        );
+        clock.tick(5100);
+      } finally {
+        clock.restore();
+      }
+    });
+
+    it('click the last one successfully with multiple clickable elements', async () => {
+      const {page} = await getTestState();
+
+      await page.setViewport({width: 500, height: 500});
+      await page.setContent(`
+        <button id="test1">test1</button>
+        <div id="test2"></div>
+        <div id="test4"></div>
+         <script>
+          const button1 = document.getElementById("test1");
+          button1.onclick = () => {
+            const div2 = document.getElementById("test4");
+            div2.innerText = "test4";
+          };
+          setTimeout(() => {
```

Comment:
instead of having setTimeout in the page context let's insert another element from the test:

``` 
await page.setContent(`<button id="test1" onclick="this.innerText = 'clicked1'">test1</button>`);

const clickDone = Locator.last([
        page.locator('#test1'),
        page.locator('#test2'),
      ]).click();

await new Promise(resolve => setTimeout(resolve, 50));

await page.evaluate(() => {
  document.body.innerHTML += `<button id="test2" onclick="this.innerText = 'clicked2'">test2</button>`'
});

await clickDone;

expect(await page.$eval('test1', (el) => el.innerText)).toBe('test1');
expect(await page.$eval('test12, (el) => el.innerText)).toBe('clicked');
```